src/modules/fundingManager/bondingCurveFundingManager/BancorVirtualSupplyBondingCurveFundingManager.sol

```diff
@@ -216,6 +227,71 @@ contract BancorVirtualSupplyBondingCurveFundingManager is
         return reserveRatioForBuying;
     }
 
+    /// @notice Calculates and returns the static price for buying the issuance token.
+    /// The return value is formatted in PPM.
+    /// @return uint The static price for buying the issuance token
+    function getStaticPriceForBuying()
+        external
+        view
+        override(BondingCurveFundingManagerBase)
+        returns (uint)
+    {
+        return _staticPricePPM(
+            virtualTokenSupply, virtualCollateralSupply, reserveRatioForBuying
+        );
+    }
+
+    /// @notice Calculates and returns the static price for selling the issuance token.
+    /// The return value is formatted in PPM.
+    /// @return uint The static price for selling the issuance token
+    function getStaticPriceForSelling()
+        external
+        view
+        override(RedeemingBondingCurveFundingManagerBase)
+        returns (uint)
+    {
+        return _staticPricePPM(
+            virtualTokenSupply, virtualCollateralSupply, reserveRatioForSelling
+        );
+    }
+
+    /// @inheritdoc IBancorVirtualSupplyBondingCurveFundingManager
+    function calculatePurchaseReturn(uint _depositAmount)
+        external
+        view
+        returns (uint mintAmount)
+    {
+        if (_depositAmount == 0) {
+            revert
+                BancorVirtualSupplyBondingCurveFundingManager__InvalidDepositAmount(
+            );
+        }
+        if (buyFee > 0) {
+            (_depositAmount, /* feeAmount */ ) =
+                _calculateNetAmountAndFee(_depositAmount, buyFee);
+        }
+        return _issueTokensFormulaWrapper(_depositAmount);
+    }
+
+    /// @inheritdoc IBancorVirtualSupplyBondingCurveFundingManager
+    function calculateSaleReturn(uint _depositAmount)
+        external
+        view
+        returns (uint redeemAmount)
+    {
+        if (_depositAmount == 0) {
+            revert
```

Comment:
Should a purely informative function really fail here?
Couldnt you just return 0 here?
This also counts for ->calculatePurchaseReturn

Reply:
If we don't, it will revert when calling the formula anyway in line 287. This way we can give an error message